app/src/main/java/de/tum/in/tumcampusapp/utils/camera/CameraManager.kt

```diff
@@ -0,0 +1,228 @@
+                openGallery()
+            }
+        }
+    }
+
+    /**
+     * @return true if user has given permission before
+     */
+    @RequiresApi(api = Build.VERSION_CODES.M)
+    private fun checkPermission(permission: String): Boolean {
+        val permissionCheck = ContextCompat.checkSelfPermission(context, permission)
+        if (permissionCheck == PackageManager.PERMISSION_DENIED) {
+            permissionLauncher.launch(arrayOf(permission))
+            return false
+        }
+        return true
+    }
+
+    private fun takePicture() {
+        // Create the file where the photo should go
+        var photoFile: File? = null
+        try {
+            try {
+                photoFile = ImageUtils.createImageFile(context)
+                currentPhotoPath = photoFile.absolutePath
+            } catch (e: IOException) {
+                Utils.log(e)
+            }
+
+            if (photoFile == null) {
+                return
+            }
+
+            val takePictureIntent = Intent(MediaStore.ACTION_IMAGE_CAPTURE)
+            val authority = "de.tum.in.tumcampusapp.fileprovider"
+            val photoURI = FileProvider.getUriForFile(context, authority, photoFile)
+            takePictureIntent.putExtra(MediaStore.EXTRA_OUTPUT, photoURI)
+            photoLauncher.launch(takePictureIntent)
+        } catch (e: ActivityNotFoundException) {
+            photoFile?.delete()
+        }
+    }
+
+    private fun openGallery() {
+        val intent = Intent()
+        intent.type = "image/*"
+        intent.action = Intent.ACTION_GET_CONTENT
+
+        try {
+            galleryLauncher.launch(Intent.createChooser(intent, "Select file"))
+        } catch (e: ActivityNotFoundException) {
+            Toast.makeText(context, R.string.error_unknown, Toast.LENGTH_SHORT).show()
+        }
+    }
+
+    private fun removeImage(path: String) {
+        val index = imageElement.indexOf(path)
+        imageElement.remove(path)
+        File(path).delete()
+        thumbnailsAdapter.removeImage(index)
+    }
+
+    private fun onNewImageTaken() {
+        currentPhotoPath?.let {
+            if (ImageUtils.rescaleBitmap(parent, it)) {
+                imageElement.add(it)
+                thumbnailsAdapter.addImage(it)
+            }
+        }
+    }
+
+    private fun onNewImageSelected(result: ActivityResult) {
+        val filePath = ImageUtils.rescaleBitmap(context, result.data?.data ?: return) ?: return
+        imageElement.add(filePath)
+        thumbnailsAdapter.addImage(filePath)
+    }
+
+    override fun clearImages() {
+        for (path in imageElement) {
+            File(path).delete()
+        }
+    }
+
+    override fun getImagePaths(): Array<String> {
+        return imageElement.toTypedArray()
+    }
+
+    private fun showThumbnailRemovedDialog(path: String) {
+        val builder = AlertDialog.Builder(parent)
```

Comment:
Consider using `de.tum.in.tumcampusapp.utils.ThemedAlertDialogBuilder` instead. AlertDialog may sometimes not have buttons visible

If this is not possible, add a comment why